pkg/cmd/cli/uninstall/uninstall.go

```diff
@@ -19,96 +19,157 @@ package uninstall
+			if !o.force {
+				fmt.Println("You are about to uninstall Velero.")
+				if !cli.GetConfirmation() {
+					// Don't do anything unless we get confirmation
+					return
+				}
+			}
+
+			client, extCl, err := kube.GetClusterClient()
+			cmd.CheckError(err)
+			cmd.CheckError(Run(context.Background(), client, extCl, f.Namespace(), o.wait))
+		},
 	}
 
-	rolebinding := install.ClusterRoleBinding(veleroNamespace)
+	o.BindFlags(c.Flags())
+	return c
+}
+
+// Run removes all components that were deployed using the Velero install command
+func Run(ctx context.Context, client *kubernetes.Clientset, extensionsClient *apiextensionsclientset.Clientset, namespace string, waitToTerminate bool) error {
+	var errs []error
 
-	err = client.RbacV1().ClusterRoleBindings().Delete(ctx, rolebinding.Name, metav1.DeleteOptions{})
+	// namespace
+	ns, err := client.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
 	if err != nil {
-		return errors.WithMessagef(err, "Uninstall failed removing Velero cluster role binding %s", rolebinding)
+		if apierrors.IsNotFound(err) {
+			fmt.Printf("Velero installation namespace %q does not exist, skipping.\n", namespace)
+		} else {
+			errs = append(errs, errors.WithStack(err))
+		}
+	} else {
+		if ns.Status.Phase == corev1.NamespaceTerminating {
+			fmt.Printf("Velero installation namespace %q is terminating.\n", namespace)
+		} else {
+			err = client.CoreV1().Namespaces().Delete(ctx, ns.Name, metav1.DeleteOptions{})
+			if err != nil {
+				errs = append(errs, errors.WithStack(err))
+			}
+		}
 	}
-	veleroLabels := labels.FormatLabels(install.Labels())
 
+	// rolebinding
+	crb := install.ClusterRoleBinding(namespace)
+	if err := client.RbacV1().ClusterRoleBindings().Delete(ctx, crb.Name, metav1.DeleteOptions{}); err != nil {
+		if apierrors.IsNotFound(err) {
+			fmt.Printf("Velero installation rolebinding %q does not exist, skipping.\n", crb.Name)
+		} else {
+			errs = append(errs, errors.WithStack(err))
+		}
+	}
+
+	// CRDs
+	veleroLabels := labels.FormatLabels(install.Labels())
 	crds, err := extensionsClient.ApiextensionsV1().CustomResourceDefinitions().List(ctx, metav1.ListOptions{
 		LabelSelector: veleroLabels,
 	})
 	if err != nil {
-		return errors.WithMessagef(err, "Uninstall failed listing Velero crds")
+		errs = append(errs, errors.WithStack(err))
 	}
-	for _, removeCRD := range crds.Items {
-		err = extensionsClient.ApiextensionsV1().CustomResourceDefinitions().Delete(ctx, removeCRD.ObjectMeta.Name, metav1.DeleteOptions{})
-		if err != nil {
-			return errors.WithMessagef(err, "Uninstall failed removing CRD %s", removeCRD.ObjectMeta.Name)
+	if len(crds.Items) == 0 {
+		fmt.Print("Velero CRDs do not exist, skipping.\n")
+	} else {
+		for _, removeCRD := range crds.Items {
+			if err = extensionsClient.ApiextensionsV1().CustomResourceDefinitions().Delete(ctx, removeCRD.ObjectMeta.Name, metav1.DeleteOptions{}); err != nil {
+				errs = append(errs, errors.WithStack(err))
```

Comment:
Let's keep the message so we know which CRD it failed on.

"Uninstall failed removing CRD %s", removeCRD.ObjectMeta.Name